async-raft/src/replication/mod.rs

```diff
@@ -767,7 +777,6 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
     #[tracing::instrument(level = "trace", skip(self, snapshot))]
     async fn stream_snapshot(&mut self, mut snapshot: CurrentSnapshotData<S::Snapshot>) -> RaftResult<()> {
         let mut offset = 0;
-        self.core.last_log_index = snapshot.index;
```

Comment:
@xu-cheng this was the core of the issue over in #98. This value should not have been updated here, not sure why I was doing so. `self.core.last_log_index` has nothing to do with a peer node's state, it only indicates that last log index to be appended to the local log by the leader (these replication streams only exist when the node is the leader).

This had the potential to cause a mismatch of data during replication, typically right after or during the process of streaming a snapshot to a follower node.